From 8f71a56267e69417014b0c22f44ad931bc553fe3 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 21 May 2021 15:23:34 -0500
Subject: [PATCH] Disable rtkit support

This can be removed if/when rtkit is packaged for Solus
---
 src/daemon/client-rt.conf.in                  | 79 -------------------
 src/daemon/filter-chain/demonic.conf          | 10 ---
 .../filter-chain/sink-dolby-surround.conf     |  9 ---
 src/daemon/filter-chain/sink-eq6.conf         |  9 ---
 .../filter-chain/sink-matrix-spatialiser.conf |  9 ---
 src/daemon/filter-chain/source-rnnoise.conf   |  9 ---
 src/daemon/jack.conf.in                       | 10 ---
 src/daemon/media-session.d/media-session.conf | 10 ---
 src/daemon/meson.build                        |  1 -
 src/daemon/pipewire-pulse.conf.in             |  9 ---
 src/daemon/pipewire.conf.in                   | 10 ---
 11 files changed, 165 deletions(-)
 delete mode 100644 src/daemon/client-rt.conf.in

diff --git a/src/daemon/client-rt.conf.in b/src/daemon/client-rt.conf.in
deleted file mode 100644
index 7bb1470f5..000000000
--- a/src/daemon/client-rt.conf.in
+++ /dev/null
@@ -1,79 +0,0 @@
-# Real-time Client config file for PipeWire version @VERSION@ #
-
-context.properties = {
-    ## Configure properties in the system.
-    #mem.warn-mlock  = false
-    #mem.allow-mlock = true
-    #mem.mlock-all   = false
-    log.level        = 0
-}
-
-context.spa-libs = {
-    #<factory-name regex> = <library-name>
-    #
-    # Used to find spa factory names. It maps an spa factory name
-    # regular expression to a library name that should contain
-    # that factory.
-    #
-    audio.convert.* = audioconvert/libspa-audioconvert
-    support.*       = support/libspa-support
-}
-
-context.modules = [
-    #{   name = <module-name>
-    #    [ args = { <key> = <value> ... } ]
-    #    [ flags = [ [ ifexists ] [ nofail ] ]
-    #}
-    #
-    # Loads a module with the given parameters.
-    # If ifexists is given, the module is ignored when it is not found.
-    # If nofail is given, module initialization failures are ignored.
-    #
-    # Uses RTKit to boost the data thread priority.
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
-
-    # The native communication protocol.
-    {   name = libpipewire-module-protocol-native }
-
-    # Allows creating nodes that run in the context of the
-    # client. Is used by all clients that want to provide
-    # data to PipeWire.
-    {   name = libpipewire-module-client-node }
-
-    # Allows creating devices that run in the context of the
-    # client. Is used by the session manager.
-    {   name = libpipewire-module-client-device }
-
-    # Makes a factory for wrapping nodes in an adapter with a
-    # converter and resampler.
-    {   name = libpipewire-module-adapter }
-
-    # Allows applications to create metadata objects. It creates
-    # a factory for Metadata objects.
-    {   name = libpipewire-module-metadata }
-
-    # Provides factories to make session manager objects.
-    {   name = libpipewire-module-session-manager }
-]
-
-filter.properties = {
-    #node.latency = 1024/48000
-}
-
-stream.properties = {
-    #node.latency = 1024/48000
-    #node.autoconnect = true
-    #resample.quality = 4
-    #channelmix.normalize = false
-    #channelmix.mix-lfe = true
-    #channelmix.upmix = false
-    #channelmix.lfe-cutoff = 0
-}
diff --git a/src/daemon/filter-chain/demonic.conf b/src/daemon/filter-chain/demonic.conf
index ffe3ea83a..94f9cf91b 100644
--- a/src/daemon/filter-chain/demonic.conf
+++ b/src/daemon/filter-chain/demonic.conf
@@ -19,16 +19,6 @@ context.spa-libs = {
 }
 
 context.modules = [
-    # Uses RTKit to boost the data thread priority.
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
 
     # The native communication protocol.
     {   name = libpipewire-module-protocol-native }
diff --git a/src/daemon/filter-chain/sink-dolby-surround.conf b/src/daemon/filter-chain/sink-dolby-surround.conf
index 943959266..e0fb22eb2 100644
--- a/src/daemon/filter-chain/sink-dolby-surround.conf
+++ b/src/daemon/filter-chain/sink-dolby-surround.conf
@@ -12,15 +12,6 @@ context.spa-libs = {
 }
 
 context.modules = [
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
     {   name = libpipewire-module-protocol-native }
     {   name = libpipewire-module-client-node }
     {   name = libpipewire-module-adapter }
diff --git a/src/daemon/filter-chain/sink-eq6.conf b/src/daemon/filter-chain/sink-eq6.conf
index 212c375bc..21b9d16f1 100644
--- a/src/daemon/filter-chain/sink-eq6.conf
+++ b/src/daemon/filter-chain/sink-eq6.conf
@@ -12,15 +12,6 @@ context.spa-libs = {
 }
 
 context.modules = [
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
     {   name = libpipewire-module-protocol-native }
     {   name = libpipewire-module-client-node }
     {   name = libpipewire-module-adapter }
diff --git a/src/daemon/filter-chain/sink-matrix-spatialiser.conf b/src/daemon/filter-chain/sink-matrix-spatialiser.conf
index 53e3b571f..daf707d53 100644
--- a/src/daemon/filter-chain/sink-matrix-spatialiser.conf
+++ b/src/daemon/filter-chain/sink-matrix-spatialiser.conf
@@ -13,15 +13,6 @@ context.spa-libs = {
 }
 
 context.modules = [
-	{   name = libpipewire-module-rtkit
-		args = {
-			#nice.level   = -11
-			#rt.prio	  = 88
-			#rt.time.soft = 200000
-			#rt.time.hard = 200000
-		}
-		flags = [ ifexists nofail ]
-	}
 	{   name = libpipewire-module-protocol-native }
 	{   name = libpipewire-module-client-node }
 	{   name = libpipewire-module-adapter }
diff --git a/src/daemon/filter-chain/source-rnnoise.conf b/src/daemon/filter-chain/source-rnnoise.conf
index 63f6d65fa..45b8e1098 100644
--- a/src/daemon/filter-chain/source-rnnoise.conf
+++ b/src/daemon/filter-chain/source-rnnoise.conf
@@ -12,15 +12,6 @@ context.spa-libs = {
 }
 
 context.modules = [
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
     {   name = libpipewire-module-protocol-native }
     {   name = libpipewire-module-client-node }
     {   name = libpipewire-module-adapter }
diff --git a/src/daemon/jack.conf.in b/src/daemon/jack.conf.in
index d7e689940..9f4f5c902 100644
--- a/src/daemon/jack.conf.in
+++ b/src/daemon/jack.conf.in
@@ -29,16 +29,6 @@ context.modules = [
     # If nofail is given, module initialization failures are ignored.
     #
     #
-    # Uses RTKit to boost the data thread priority.
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
 
     # The native communication protocol.
     {   name = libpipewire-module-protocol-native }
diff --git a/src/daemon/media-session.d/media-session.conf b/src/daemon/media-session.d/media-session.conf
index 914d9154f..7b934985a 100644
--- a/src/daemon/media-session.d/media-session.conf
+++ b/src/daemon/media-session.d/media-session.conf
@@ -27,16 +27,6 @@ context.modules = [
     # If ifexists is given, the module is ignored when it is not found.
     # If nofail is given, module initialization failures are ignored.
     #
-    # Uses RTKit to boost the data thread priority.
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
 
     # The native communication protocol.
     {   name = libpipewire-module-protocol-native }
diff --git a/src/daemon/meson.build b/src/daemon/meson.build
index 0429698a8..1b66980c3 100644
--- a/src/daemon/meson.build
+++ b/src/daemon/meson.build
@@ -28,7 +28,6 @@ conf_config_uninstalled.set('comment', '')
 conf_files = [
   [ 'pipewire.conf.in', 'pipewire.conf' ],
   [ 'client.conf.in', 'client.conf' ],
-  [ 'client-rt.conf.in', 'client-rt.conf' ],
   [ 'jack.conf.in', 'jack.conf' ],
   [ 'pipewire-pulse.conf.in', 'pipewire-pulse.conf' ]
 ]
diff --git a/src/daemon/pipewire-pulse.conf.in b/src/daemon/pipewire-pulse.conf.in
index c481684ef..849909e90 100644
--- a/src/daemon/pipewire-pulse.conf.in
+++ b/src/daemon/pipewire-pulse.conf.in
@@ -14,15 +14,6 @@ context.spa-libs = {
 }
 
 context.modules = [
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
     {   name = libpipewire-module-protocol-native }
     {   name = libpipewire-module-client-node }
     {   name = libpipewire-module-adapter }
diff --git a/src/daemon/pipewire.conf.in b/src/daemon/pipewire.conf.in
index bbafa134b..a23803813 100644
--- a/src/daemon/pipewire.conf.in
+++ b/src/daemon/pipewire.conf.in
@@ -62,16 +62,6 @@ context.modules = [
     # If nofail is given, module initialization failures are ignored.
     #
 
-    # Uses RTKit to boost the data thread priority.
-    {   name = libpipewire-module-rtkit
-        args = {
-            #nice.level   = -11
-            #rt.prio      = 88
-            #rt.time.soft = 200000
-            #rt.time.hard = 200000
-        }
-        flags = [ ifexists nofail ]
-    }
 
     # The native communication protocol.
     {   name = libpipewire-module-protocol-native }
-- 
2.30.2

